src/commonmark.rs

```diff
@@ -440,6 +440,68 @@ impl<'a, W: Write + 'a> Context<'a, W> {
         );
         write!(self.output.writer, "{}", osc(&command))
     }
+
+    /// Write an inline image for Terminology.
+    fn write_terminology_inline_image(&mut self, path: &Path) -> Result<()> {
+        fn get_lines(columns: u16, file: &str) -> u16 {
+            use std::process::Command;
+            use termion::terminal_size;
+
+            let lines = terminal_size().unwrap_or((0, 24)).1;
+
+            Command::new("identify")
+                .arg("-format")
+                .arg("%w %h")
+                .arg(file)
+                .output()
+                .and_then(|o| {
+                    fn get_v(x: Option<&str>) -> Result<u32> {
+                        x.and_then(|x| x.parse().ok())
+                            .ok_or(Into::<Error>::into(ErrorKind::InvalidData))
+                    }
+
+                    let o = String::from_utf8(o.stdout).or(Err(ErrorKind::InvalidData))?;
+                    let mut tokens = o.split(' ');
+                    let (w, h) = (get_v(tokens.next())? as f64, get_v(tokens.next())? as f64);
+
+                    // We divide by 2 because terminal cursor/font usually has a
+                    // 1:2 proportion
+                    Ok((h * (columns / 2) as f64 / w) as u16)
+                })
+                .unwrap_or(lines / 2)
+        }
+
+        fn is_http_url(path: &Path) -> bool {
+            path.to_str()
+                .map(|p| p.starts_with("http"))
+                .unwrap_or(false)
+        }
+
+        let arg: Cow<str> = if is_http_url(path) {
+            From::from(path.to_str().ok_or(ErrorKind::InvalidData)?)
+        } else if path.is_relative() {
+            From::from(self.input
+                .base_dir
+                .join(path)
+                .to_str()
+                .map(str::to_owned)
+                .ok_or(ErrorKind::InvalidData)?)
+        } else {
+            From::from(path.to_str().ok_or(ErrorKind::InvalidData)?)
```

Comment:
This `else` is the same as the first `if`, isn't it?  Please refactor this condition to avoid the duplication.

Reply:
That said, you don't need to check whether path is relative; joining a path with an absolute path just returns the latter, so you can just use `self.input.base_dir.join(path)` in all cases, and it'll always give you a correct path.

Reply:
> This else is the same as the first if, isn't it?

You're right. Gonna do.

> so you can just use self.input.base_dir.join(path) in all cases, and it'll always give you a correct path

If path is already absolute, I don't want to change the path by joining.

Reply:
You won't change the path, I think; I prefer having a simpler code path with less conditions.

Reply:
> You won't change the path, I think

You're right: https://play.rust-lang.org/?gist=5651ff43abd5f8bc7af2eaeb1c0cde36&version=stable

But I still need to have different code path for remote URLs.